tests/ga/test_update.py

```diff
@@ -1670,15 +1673,85 @@ def _mock_popen(cmd, *args, **kwargs):
         self.assertEqual(PersistFirewallRulesHandler._FIREWALLD_RUNNING_CMD, executed_commands.pop(0),
                          "First command should be to check if firewalld is running")
         self.assertTrue([FirewallCmdDirectCommands.QueryPassThrough in cmd for cmd in executed_commands],
                         "The remaining commands should only be for querying the firewall commands")
 
+    def test_it_should_set_dns_tcp_iptable_if_drop_available_accept_unavailable(self):
+        osutil._enable_firewall = True
+
+        with TestOSUtil._mock_iptables() as mock_iptables:
+            with _get_update_handler(test_data=DATA_FILE) as (update_handler, _):
+                # drop rule is present
+                mock_iptables.set_command(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination), exit_code=0)
+                # non root tcp iptable rule is absent
+                mock_iptables.set_command(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination), exit_code=1)
+                update_handler.run(debug=True)
+
+                drop_check_command = TestOSUtil._command_to_string(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination))
+                accept_tcp_check_rule = TestOSUtil._command_to_string(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination))
+                accept_tcp_insert_rule = TestOSUtil._command_to_string(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.INSERT_COMMAND, mock_iptables.destination))
+
+                get_firewall_packets = TestOSUtil._command_to_string(osutil.get_firewall_packets_command(mock_iptables.wait))
+
+                self.assertEqual(len(mock_iptables.command_calls), 4,"Incorrect number of calls to iptables: [{0}]".format(mock_iptables.command_calls))
+                self.assertEqual(mock_iptables.command_calls[0], drop_check_command,
+                                 "The first command should check the drop rule")
+                self.assertEqual(mock_iptables.command_calls[1], accept_tcp_check_rule,
+                                 "The second command should check the accept rule")
+                self.assertEqual(mock_iptables.command_calls[2], accept_tcp_insert_rule,
+                                 "The third command should add the accept rule")
+                self.assertEqual(mock_iptables.command_calls[3], get_firewall_packets,
+                                 "The fourth command should be for listing the firewall rules")
+
+    def test_it_should_not_set_dns_tcp_iptable_if_drop_unavailable(self):
+        osutil._enable_firewall = True
+
+        with TestOSUtil._mock_iptables() as mock_iptables:
+            with _get_update_handler(test_data=DATA_FILE) as (update_handler, _):
+                #drop rule is not available
```

Comment:
NIT: Pep 8 warning - 
> PEP 8: E265 block comment should start with '# '

(on the side note, please enable PEP8 inspection in pycharm :) )

Reply:
Addressed. Hopefully I did not miss anything more this time :)